deeplabcut/pose_estimation_tensorflow/predict_videos.py

```diff
@@ -1798,26 +1799,31 @@ def convert_detections2tracklets(
                     )
                 tracklets = {}
                 multi_bpts = cfg["multianimalbodyparts"]
-                ass = inferenceutils.Assembler(
-                    data,
-                    max_n_individuals=inferencecfg["topktoretain"],
-                    n_multibodyparts=len(multi_bpts),
-                    greedy=greedy,
-                    pcutoff=inferencecfg.get("pcutoff", 0.1),
-                    min_affinity=inferencecfg.get("pafthreshold", 0.05),
-                    window_size=window_size,
-                    identity_only=identity_only,
-                )
-                if calibrate:
-                    trainingsetfolder = auxiliaryfunctions.get_training_set_folder(cfg)
-                    train_data_file = os.path.join(
-                        cfg["project_path"],
-                        str(trainingsetfolder),
-                        "CollectedData_" + cfg["scorer"] + ".h5",
+                if not os.path.exists(dataname.split(".h5")[0] + "_assemblies.pickle") or overwrite:
+                    ass = inferenceutils.Assembler(
+                        data,
+                        max_n_individuals=inferencecfg["topktoretain"],
+                        n_multibodyparts=len(multi_bpts),
+                        greedy=greedy,
+                        pcutoff=inferencecfg.get("pcutoff", 0.1),
+                        min_affinity=inferencecfg.get("pafthreshold", 0.05),
+                        window_size=window_size,
+                        identity_only=identity_only,
                     )
-                    ass.calibrate(train_data_file)
-                ass.assemble()
-                ass.to_pickle(dataname.split(".h5")[0] + "_assemblies.pickle")
+                    if calibrate:
+                        trainingsetfolder = auxiliaryfunctions.get_training_set_folder(cfg)
+                        train_data_file = os.path.join(
+                            cfg["project_path"],
+                            str(trainingsetfolder),
+                            "CollectedData_" + cfg["scorer"] + ".h5",
+                        )
+                        ass.calibrate(train_data_file)
+                    ass.assemble()
+                    ass.to_pickle(dataname.split(".h5")[0] + "_assemblies.pickle")
+                else:
+                    ass_filename = dataname.split(".h5")[0] + "_assemblies.pickle"
+                    print(f"Loading assemblies from {ass_filename}")
+                    ass = pd.read_pickle(dataname.split(".h5")[0] + "_assemblies.pickle")
```

Comment:
@KonradDanielewski, I'd be in favor of adding a `classmethod` called `Assembler.from_pickle`, as another constructor: what would you think about that? We'd encapsulate the code you wrote into that function, so we manipulate early on an `Assembler` object only, simplifying the code further below.

Reply:
Sure, that makes more sense. I'll try to get it done today.

Reply:
Like we discussed, because the `full.pickle` is loaded anyway, we just initialize the Assembler and then fill the assemblies and unique from the file instead of creating a `classsmethod`. This avoids the necessity of creating fake metadata etc.